alb/alb_test.go

```diff
@@ -108,27 +108,20 @@ func (m *mockMetadata) mockInstanceMetadata(instanceID string) {
 }
 
 func setup(targetGroupNames ...string) (controller.Updater, *mockALB, *mockMetadata) {
-	a := New(region, targetGroupNames, time.Nanosecond)
+	a, _ := New(region, targetGroupNames, time.Nanosecond)
 	mockALB := &mockALB{}
 	mockMetadata := &mockMetadata{}
 	a.(*alb).awsALB = mockALB
 	a.(*alb).metadata = mockMetadata
 	return a, mockALB, mockMetadata
 }
 
-func TestNoopIfNoExpectedFrontEnds(t *testing.T) {
-	//given
-	a, mockElb, mockMetadata := setup()
-	a.(*alb).targetGroupNames = nil
-
+func TestCanNotCreateUpdaterWithoutLabelValue(t *testing.T) {
 	//when
-	a.Start()
-	a.Update(controller.IngressEntries{})
-	a.Stop()
+	_, err := New(region, []string{}, time.Nanosecond)
 
 	//then
-	mockElb.AssertExpectations(t)
-	mockMetadata.AssertExpectations(t)
+	assert.EqualError(t, err, "Unable to create Alb Updater: missing target group names")
```

Comment:
I don't think we should assert against the exact error message, just assert the error is not nil.